### ISSUE TITLE
About the padding size of the Series Informed Activation Function

### ISSUE THREAD

**Issue by 66Kevin** [huawei-noah/VanillaNet#4]
Hi,
Thanks for your amazing work!
I found the [padding size](https://github.com/huawei-noah/VanillaNet/blob/f8385cf4ac0a5143431a05c84c5a40da4491b3d8/models/vanillanet.py#L29) used in activation funtion is actually same as `padding=self.act_num`, so can i ask why you use `padding=(self.act_num*2+1)//2` instead of `padding=self.act_num`? It seems could both guarded the input size is equal to output size after the DW-conv.

Best,
Yueyi

**Comment by HantingChen** [comment]
Thank for the nice suggestion! We have fixed this.

**Comment by 66Kevin** [comment]
> Thank for the nice suggestion! We have fixed this.

Thanks for your reply! There is also a question about Series Informed Activation Function. Why the number of stacked activation function can be implemented in act_num in the following code snippet? It seems only enlarge the kernel size and how to achieve the goal of stacking activtion function? Thanks for your patient!
`self.weight = torch.nn.Parameter(torch.randn(dim, 1, act_num * 2 + 1, act_num * 2 + 1))`

Best,
Yueyi

**Comment by HantingChen** [comment]
Thanks for your question! By enlarging the kernel size in the code, we are effectively aggregating outputs from the activation function at different positions. This essentially achieves the stacking of the activation function. Although it may seem counterintuitive at first, it is simply an implementation strategy to achieve our goal. Please feel free to ask if you have more questions!
